mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td

```diff
@@ -2125,30 +2125,41 @@ def TensorStoreOp : MemRef_Op<"tensor_store",
 def MemRef_TransposeOp : MemRef_Op<"transpose", [
     DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
     Pure]>,
-    Arguments<(ins AnyStridedMemRef:$in, AffineMapAttr:$permutation)>,
+    Arguments<(ins AnyStridedMemRef:$in, DenseI64ArrayAttr:$permutation)>,
```

Comment:
Not a big deal, but DenseI32ArrayAttr (or even DenseI16ArrayAttr!) are quite likely enough